acquia/cli
# New command: `app:new:from:drupal7`

**Motivation**
ACLI already has a `app:new:local` command that spins up a brand new Drupal project.

Wouldn't it be great if it also helped you spin up a new Drupal 9+ project that helps migrating from Drupal 7?

**Proposed changes**
Introduce `acli app:new:from:drupal7`.

All code has been battle-hardened over years of use in production on Acquia Cloud. Please check the commits in sequence, because I designed them to tell a gradual story:
1. code to inspect a Drupal 7 site's installed modules
2. introduce the new ACLI command
3. add unit tests
4. add integration tests

Until the full set of _recommendations_ is released, this will use only a minimal set of recommendations, primarily consisting of some Drupal 7 → 9 migration recommendations for Acquia modules.

This is a necessary step in the way of open sourcing https://www.acquia.com/products/drupal-cloud/acquia-migrate-accelerate.

**Alternatives considered**
N/A

**Testing steps**
Point it at a Drupal 7 site and observe that it generates a valid `composer.json` that results in an installable Drupal 9+ site. See the integration tests for various sample Drupal 7 sites at `tests/phpunit/src/Commands/App/NewFromDrupal7CommandTest.php`, with fixtures at `tests/fixtures/drupal7`.
